interpreter.go

```diff
@@ -372,7 +389,8 @@ func (i *interpreter) evaluate(a ast.Node) (value, error) {
 				return nil, err
 			}
 			// TODO(https://github.com/google/jsonnet/issues/377): non-integer indexes should be an error
-			return e.evaluate(target.elements[int(indexInt.value)])
+			return e.evaluateTailCall(target.elements[int(indexInt.value)], tc)
```

Comment:
I think the C++ implementation would use nonTailCall here, but actually this does make sense